src/Zypper.cc

```diff
@@ -1220,10 +1222,17 @@ void Zypper::processGlobalOptions()
     MIL << "gpg-auto-import-keys is on" << endl;
   }
 
-  if ( (it = gopts.find("root")) != gopts.end() )
+  if ( (it = gopts.find("root")) != gopts.end() || (it = gopts.find("installroot")) != gopts.end() )
   {
+    if ( gopts.find("root") != gopts.end() && gopts.find("installroot") != gopts.end() )
+    {
+      out().error(_("--root can not be used together with the --installroot flag."));
+      setExitCode( ZYPPER_EXIT_ERR_INVALID_ARGS );
+      return;
```

Comment:
I know its 'Kraut und Rüben' in the code. The desired style until the error handling is redesigned is throwing an ExitRequestException:
```
out().error(_("...") );
setExitCode( ZYPPER_EXIT_ERR_INVALID_ARGS );
ZYPP_THROW( ExitRequestException("invalid args") );
```

Two rules for the translated error messages as we (want to) avoid duplication:
 - Fix strings like option names should be injected via Format. I.e use one translated string for multiple cases.
 - If the same translated string is used multiple times and at multiple places, we should try to use a function.  providing/building the string. I recently created a 'cli' namespace in Zypper.cc to collect all such stuff. If you rebase on 67ad041420186ca54597dbffb050905efa5dc875 you can use this message:

```
  out().error( cli::errorMutuallyExclusiveOptions( "--root --installroot" ) );
  ...
```

Reply:
Done